pkg/apis/pipeline/v1beta1/pipeline_types_test.go

```diff
@@ -1139,3 +1129,149 @@ func TestPipelineTask_IsMatrixed(t *testing.T) {
+		{
+			name: "matrixed with params",
+			arg: arg{
+				Matrix: &Matrix{
+					Params: []Param{{Name: "platform", Value: ParamValue{ArrayVal: []string{"linux", "windows"}}}},
+				},
+			},
+			expected: true,
+		}, {
+			name: "matrixed with include",
+			arg: arg{
+				Matrix: &Matrix{
+					Include: []MatrixInclude{{
+						Name: "build-1",
+						Params: []Param{{
+							Name: "IMAGE", Value: ParamValue{Type: ParamTypeString, StringVal: "image-1"},
+						}, {
+							Name: "DOCKERFILE", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/Dockerfile1"}}},
+					}},
+				},
+			},
+			expected: false,
+		}, {
+			name: "matrixed with params and include",
+			arg: arg{
+				Matrix: &Matrix{
+					Params: []Param{{
+						Name: "GOARCH", Value: ParamValue{ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+					}},
+					Include: []MatrixInclude{{
+						Name: "common-package",
+						Params: []Param{{
+							Name: "package", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"}}},
+					}},
+				},
+			},
+			expected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			matrix := tc.arg.Matrix
+			matrixHasParams := matrix.MatrixHasParams()
+			if matrixHasParams != tc.expected {
+				t.Errorf("Matrix.MatrixHasParams() return bool: %v, but wanted: %v", matrixHasParams, tc.expected)
+			}
+		})
+	}
+}
+
+func TestPipelineTask_MatrixHasInclude(t *testing.T) {
+	type arg struct {
+		*Matrix
+	}
+	testCases := []struct {
+		name     string
+		arg      arg
+		expected bool
+	}{
+		{
+			name: "nil matrix",
+			arg: arg{
+				Matrix: nil,
+			},
+			expected: false,
+		},
+		{
+			name: "empty matrix",
+			arg: arg{
+				Matrix: &Matrix{},
+			},
+			expected: false,
+		},
+		{
+			name: "matrixed with params",
+			arg: arg{
+				Matrix: &Matrix{
+					Params: []Param{{Name: "platform", Value: ParamValue{ArrayVal: []string{"linux", "windows"}}}},
+				},
+			},
+			expected: false,
+		}, {
+			name: "matrixed with include",
+			arg: arg{
+				Matrix: &Matrix{
+					Include: []MatrixInclude{{
+						Name: "build-1",
+						Params: []Param{{
+							Name: "IMAGE", Value: ParamValue{Type: ParamTypeString, StringVal: "image-1"},
+						}, {
+							Name: "DOCKERFILE", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/Dockerfile1"}}},
+					}},
+				},
+			},
+			expected: true,
+		}, {
+			name: "matrixed with params and include",
+			arg: arg{
+				Matrix: &Matrix{
+					Params: []Param{{
+						Name: "GOARCH", Value: ParamValue{ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+					}},
+					Include: []MatrixInclude{{
+						Name: "common-package",
+						Params: []Param{{
+							Name: "package", Value: ParamValue{Type: ParamTypeString, StringVal: "path/to/common/package/"}}},
+					}},
+				},
+			},
+			expected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			matrix := tc.arg.Matrix
```

Comment:
nit: this is a bit redundant, not sure a separate variable is needed

Reply:
In the validation and "FanOut()" Implementation I use these functions to easily differentiate between the 3 different use cases:
1. Only Matrix.Params
2. Only Matrix.Include
3. Both Matrix.Params and Matrix.Include

Reply:
Oh I see. Thanks!